src/WebJobs.Extensions.DocumentDB/Config/DocumentDBConfiguration.cs

```diff
@@ -19,14 +19,17 @@ public class DocumentDBConfiguration : IExtensionConfigProvider
     {
         internal const string AzureWebJobsDocumentDBConnectionStringName = "AzureWebJobsDocumentDBConnectionString";
         internal readonly ConcurrentDictionary<string, IDocumentDBService> ClientCache = new ConcurrentDictionary<string, IDocumentDBService>();
-        private string _defaultConnectionString;
 
         /// <summary>
         /// Constructs a new instance.
         /// </summary>
         public DocumentDBConfiguration()
         {
             DocumentDBServiceFactory = new DefaultDocumentDBServiceFactory();
+
+            // default the connection string
+            var nameResolver = new DefaultNameResolver();
```

Comment:
I'd rather not do this -- it breaks the DI/mockability that my change purposefully introduced. 

It's nice to be able to register your own NameResolver and know that we'd use it everywhere we did resolution. If we hardcode the DefaultNameResolver here, the story is "we'll use your registered INameResolver for all resolution except for when we fill in default values, which you cannot change."

I'd rather fix the tests in Script (which I can take care of). They were taking advantage of an implementation detail that has since changed. The behavior of real functions shouldn't be affected.

---
In reply to: [70554004](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/121#discussion_r70554004) [](ancestors = 70554004)